File: from-0401-to-0500/0452-minimum-number-of-arrows-to-burst-balloons/solution.go
func findMinArrowShots(points [][]int) int {
    sort.Slice(points, func(i, j int) bool {
        return points[i][1] < points[j][1]
    })
    arrowX := points[0][1]
    count := 1
    for _, p := range points {
        s, e := p[0], p[1]
        if s > arrowX {
            count += 1
            arrowX = e
        }
    }
    return count
}

File: from-0401-to-0500/0452-minimum-number-of-arrows-to-burst-balloons/solution.py
class Solution:
    def findMinArrowShots(self, points: List[List[int]]) -> int:
        points.sort(key=lambda x: x[1])
        arrow_x = points[0][1]
        count = 1
        for start, end in points:
            if start > arrow_x:
                count += 1
                arrow_x = end
                
        return count